src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,784 @@
+                    }
+                } catch (DukeException | DateTimeParseException e) {
+                    System.out.println("Skipping invalid task: " + line);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new DukeException("File not found: " + e.getMessage());
+        }
+        return tasks;
+
+    }
+    /**
+     * Saves the current tasks to the file.
+     *
+     * @param tasks The TaskList containing tasks to save.
+     */
+    public void save(TaskList tasks) {
+        try (PrintWriter writer = new PrintWriter(new FileWriter(filePath))) {
+            for (int i = 0; i < tasks.getSize(); i++) {
+                Task task = tasks.getTask(i);
+                writer.println(taskToFileString(task));
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving tasks to file: " + e.getMessage());
+        }
+    }
+    /**
+     * Converts a task to a formatted string for saving to the file.
+     * The format includes the task type, status, description, and any additional information.
+     *
+     * @param task The task to be converted to a string.
+     * @return A formatted string representing the task.
+     */
+    private String taskToFileString(Task task) {
+        String type = task instanceof Todo ? "T" :
+                task instanceof Deadline ? "D" :
+                        task instanceof Event ? "E" : "";
+        String status = task.isDone ? "1" : "0";
+        String details = task.getDescription();
+        String additionalInfo = "";
+
+        if (task instanceof Deadline) {
+            Deadline deadline = (Deadline) task;
+            additionalInfo = " | " + deadline.getBy().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+        } else if (task instanceof Event) {
+            Event event = (Event) task;
+            additionalInfo = " | " + event.getFrom().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")) +
+                    " | " + event.getTo().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+        }
+
+        return type + " | " + status + " | " + details + additionalInfo;
+    }
+
+}
+
+/**
+ * Handles parsing of user input commands and converting them into Command objects.
+ */
+class Parser {
+    private static final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    /**
+     * Parses user input and returns the appropriate Command object.
+     *
+     * @param fullCommand The full user input command.
+     * @return The command object corresponding to the user input.
+     * @throws DukeException If the command is invalid or if the input format is incorrect.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] commandParts = fullCommand.split(" ", 2);
+        String commandType = commandParts[0];
+        String commandArgs = commandParts.length > 1 ? commandParts[1] : "";
```

Comment:
Should we rename commandArgs as a singular as it is not a collection (yet)? Ex. commandArgLine, argString, etc.